react-components/src/components/Architecture/utilities.ts

```diff
@@ -15,12 +15,27 @@ export function getIcon(command: BaseCommand): IconName | undefined {
   return command.icon;
 }
 
-export function getFlexDirection(isHorizontal: boolean): FlexDirection {
-  return isHorizontal ? 'row' : 'column';
+export function getFlexDirection(placement: PlacementType): FlexDirection {
+  return placement === 'top' || placement === 'bottom' ? 'row' : 'column';
 }
 
-export function getTooltipPlacement(isHorizontal: boolean): PlacementType {
-  return isHorizontal ? 'top' : 'right';
+export function getDividerDirection(placement: PlacementType): string {
+  return placement === 'top' || placement === 'bottom' ? 'vertical' : 'horizontal';
+}
+
+export function getTooltipPlacement(placement: PlacementType): PlacementType {
+  switch (placement) {
+    case 'top':
+      return 'bottom';
+    case 'bottom':
+      return 'top';
+    case 'left':
+      return 'right';
+    case 'right':
+      return 'left';
+    default:
+      return 'top';
+  }
```

Comment:
Maybe a way to help this function be self-explanatory would be to change the name of the argument to something like "parentPlacement" or something like that to understand that it is getting the placement of a source and returning where the placement of the correlated tooltip would be. Because when you look at the definition of this function and what it checks and what it returns, it seems a bit ambiguous, IMO. So maybe it would be good to make it more explicit or create a comment explaining the reason for this switch and the argument.

Reply:
```
export function getTooltipPlacement(toolbarPlacement: PlacementType): PlacementType {
  // Try to keep the tooltip on the opposite side of the toolbar
  switch (toolbarPlacement) {
    case 'top':
      return 'bottom';
    case 'bottom':
      return 'top';
    case 'left':
      return 'right';
    case 'right':
      return 'left';
    default:
      return 'top';
  }
}

```